src/radical/pilot/session.py

```diff
@@ -24,36 +12,40 @@
 import radical.saga                 as rs
 import radical.saga.utils.pty_shell as rsup
 
-from . import utils         as rpu
-
 from .resource_config import ResourceConfig
 from .db              import DBSession
-from .utils           import version_detail as rp_version_detail
+from .                import utils          as rpu
 
 
 # ------------------------------------------------------------------------------
 #
 class Session(rs.Session):
-    """
-    A Session encapsulates a RADICAL-Pilot instance and is the *root* object
-
-    A Session holds :class:`radical.pilot.PilotManager` and
+    '''
+    A Session is the root object of all RP objects in an application instance:
+    it holds :class:`radical.pilot.PilotManager` and
     :class:`radical.pilot.UnitManager` instances which in turn hold
     :class:`radical.pilot.ComputePilot` and :class:`radical.pilot.ComputeUnit`
-    instances.
-    """
+    instances, and several other components which operate on those stateful
+    entities.
+    '''
+
+    # In that role, the session will create a special pubsub channel `heartbeat`
```

Comment:
I would add the following to the previous docstring so that it goes into our documentation automatically.

Reply:
Not sure - I would consider the heartbeat management to be an implementation detail, not something the end user would need to know as part of the API doc (it is nowhere exposed in the API).  Do you think differently?

Reply:
This would end up in the automatically generated documentation that is intended for developers and necessarily (only) for end users? I see the distinction between API and impl. detail but I also see the need to expose very valuable documentation about our implementation. Maybe something we can get smarter with Sphinx?